### ISSUE TITLE
Please, give us permissions to access your clipboard.

### ISSUE THREAD

**Issue by vvvamik** [photopea/photopea#6850]
Your Operating System: Windows 11 (x64)
Your Web Browser: Firefox 127
The problem happens in the Incognito Mode, too.

Photopea can't access the clipoard. In the latest version of Firefox, this can't even be set up. What could be the problem? It was working yesterday.

![photopea](https://github.com/photopea/photopea/assets/22279127/b77699e8-1c96-4f5d-afd1-c41fa7daaf3a)

**Comment by photopea** [comment]
I think you were using Firefox 126 yesterday, and you are using Firefox 127 today. So it is a different web browser.

I just tried to copy-paste in Firefox 127 and it works fine :/ I can even copy at www.Photopea.com (using Ctrl+C) and paste it into www.imgur.com (using Ctrl+V)

**Comment by vvvamik** [comment]
Thx for answer. It was also possible to directly copy (Ctrl+C, Ctrl+V) images from pc to Photopea. Not now. I'll have to get used to different ways of working, but Photopea is still a great project in terms of options.

**Comment by photopea** [comment]
Could you reinstall Firefox? What should I do to see a message that you see in your Photopea?

**Comment by vvvamik** [comment]
I really don't know. Maybe yesterday's KB5039212 update is to blame. As I wrote, I'll get used to it and just change the way I use Photopea.
![photopea](https://github.com/photopea/photopea/assets/22279127/4b81d6ff-7291-49af-a606-d0e49b613c46)

**Comment by photopea** [comment]
Could you try it in the Incognito mode in Firefox?

**Comment by Frog101** [comment]
> Could you try it in the Incognito mode in Firefox?

Still does the same thing, I'm experiencing the exact same issues :(

**Comment by vvvamik** [comment]
> Could you reinstall Firefox? What should I do to see a message that you see in your Photopea?

Yes. I also tried Firefox Nightly 129 and the same problem.

**Comment by photopea** [comment]
This article can tell you more about Firefox 127, that has been released yesterday: https://www.omgubuntu.co.uk/2024/06/firefox-127-improves-screenshot-tool-mixed-content-https#:~:text=a%20change%20to-,clipboard,-pasting%20now%20that

**Comment by friguron** [comment]
Same here, it's been this annoying for the last 1-2 months with Firefox, no matter if private tab or not...
In my case I'm using MacOS (just in case it's useful).

**Comment by vvvamik** [comment]
@Frog101 @friguron @photopea I don't know about you, but it's working normally again today. Problem solved from my point of view.

**Comment by photopea** [comment]
I was desperate about it, so I have disabled the Clipboard API for all Firefox users. Now, clipboard in Firefox 107 will work the same as a clipboard in Firefox 106.

To use a real Clipboard API in Photopea (e.g. to be able to copy images in Photopea and paste them outside Photopea), use other browsers. I will re-enable it for Firefox once Firefox fixes it.

**Comment by yoelwep13578** [comment]
I've tried enabling everything related with `clipboard` in Firefox `about:config`, then copying image in Photopea (current edit) and paste to other seems works. Tested in Firefox + ChromeMask to able lying and tell itself as Chrome.

Maybe this is nice time to re-enable Clipboard API for firefox. Add this guide below "Give us clipboard permission" text in popup to tell Firefox user to:

1. Visit `about:config`
2. Find anything related with `asyncClipboard`
3. Set them to true
